internal/deploy/base.go

```diff
@@ -44,6 +44,11 @@ type ServiceOperator interface {
 	Uninstall(ctx context.Context) error
 }
 
+// ServiceOperatorAction represents an action that can be run within a ServiceOperator
+type ServiceOperatorAction interface {
```

Comment:
Wdyt naming it as `ServiceOperation`?

Reply:
That works for me! Naming things is hard ,😄